gems/aws-sdk-s3/lib/aws-sdk-s3/plugins/access_grants.rb

```diff
@@ -80,6 +102,12 @@ def with_metric(credentials, &block)
             Aws::Plugins::UserAgent.metric('S3_ACCESS_GRANTS', &block)
           end
 
+          # HeadBucket is a supported call. When fetching credentials,
+          # this plugin is executed again, and becomes recursive.
+          def head_bucket_call?(provider)
```

Comment:
could this just be a method on the provider?

Reply:
It could but I didn't want it part of the public interface.